app/controllers/applications_controller.rb

```diff
@@ -0,0 +1,45 @@
+class ApplicationsController < ApplicationController
+  def show
+    @application = Application.find(params[:id])
+    if params[:pet_name].present?
+      @pets = Pet.search(params[:pet_name])
+    else
+      @pets = @application.pet_search(params[:pet_name])
+    end
+  end
+
+  def new 
+
+  end
+
+  def create 
+    application = Application.new({
+      name: params[:name],
+      street: params[:street],
+      city: params[:city],
+      state: params[:state],
+      zip: params[:zip],
+      descr: params[:good_home]
+    })
+
+    if application.save
+      redirect_to "/applications/#{application.id}"
+    else 
+      flash[:alert] = "Application not created: Required information missing."
```

Comment:
Nice sad path handling. Keep in mind in the future that you can use dynamic messaging for these errors by sending the specific AR error in the flash: `flash[:alert] = application.errors.full_messages`